pkg/controlplane/instance.go

```diff
@@ -567,6 +568,22 @@ func labelAPIServerHeartbeatFunc(identity string) lease.ProcessLeaseFunc {
 
 		// convenience label to easily map a lease object to a specific apiserver
 		lease.Labels[apiv1.LabelHostname] = hostname
+
+		// Include apiserver network location used by peers to proxy requests between kube-apiservers
+		if PeerProxyFeatureEnabled() {
+			var ip string
+			if peerBindAddress.PeerBindIP != "" && peerBindAddress.PeerBindPort != "" {
+				ip = fmt.Sprintf("%s.%s", peerBindAddress.PeerBindIP, peerBindAddress.PeerBindPort)
+			} else {
+				ip, err = reconciler.GetEndpoint(apiserverId)
```

Comment:
we do have this ip address without any need to get it from the endpoint, it is in `c.GenericConfig.PublicAddress` , I don't think we need to get the reconciler here

Reply:
yeah that makes sense. Updated.